cardinal/world.go

```diff
@@ -326,9 +327,10 @@ func (w *World) StartGame() error {
 		return err
 	}
 	if !w.instance.DoesEngineHaveAnEventHub() {
-		w.instance.SetEventHub(events.NewWebSocketEventHub())
+		eventHub := events.NewWebSocketEventHub()
+		w.instance.SetEventHub(eventHub)
```

Comment:
would it be possible to make it so engine just always has an event hub, so `World` doesn't have to be concnered with that?

Reply:
No. You have to emit events in a system through world context. So engine must know what an event hub is. 

Then the server layer emits the events to the users through fiber. So Server needs to know about eventhub. 

eventHub has to live as a 3rd entity: Server, Engine, EventHub. It really depends on how you organize it. If Server, owns engine, then it doesn't explicitly need an event hub it just uses engine and vice versa.

As a package anything can own events. You can put it under server or engine or even outside of cardinal.

Reply:
What im saying is - is it possible for `engine` to _always_ have an event hub, so the world doesn't have to configure one for it?

Reply:
sure but this is a new design direction. I would say it shouldn't be related to this PR. 

I am slightly against this though because eventHub is an interface. There can be multiple implementations and thus it's required to be "injected" in. If it "always" has it it will "always" have one implementation.

Reply:
no, because you still run the options, allowing different event hubs to be passed in. if no option is passed in, default to websockets, exactly as you have above. but instead of making the `World` object responsible for the default, just do it in `Engine`.

```go
	for _, opt := range opts {
		opt(e)
	}
	if e.receiptHistory == nil {
		e.receiptHistory = receipt.NewHistory(e.CurrentTick(), defaultReceiptHistorySize)
	}
	if e.eventHub == nil {
		e.eventHub = events.NewWebSocketEventHub()
	}
```

Reply:
this will allow you to get rid of the `DoesEngineHaveAnEventHub` method as well

Reply:
I'm also in favour of what Tyler is suggesting, seems to make sense to me, indifferent which PR it happens in but seems like it is small enough to fit in this one if you want to

Reply:
Let's move it to another PR. These function were in there originally.

Reply:
https://linear.app/arguslabs/issue/WORLD-750/make-engine-responsible-for-defaulting-eventhub

Reply:
> I'm also in favour of what Tyler is suggesting, seems to make sense to me, indifferent which PR it happens in but seems like it is small enough to fit in this one if you want to

SetEventHub and DoesEngineHaveAnEventHub and the whole thing was designed to have Dependency injection. These were all originally there and unrelated to this PR. We should move away from piggy backing changes into every PR because that's what leads to stuff like this server refactor side branch where just too much stuff gets into a single PR that it becomes buggy and nobody really does a thorough review of it. 

This PR is a PR into another giant feature branch that 3 people picked up and applied changes to. Then on top of that we have even more changes concurrently going into main as we speak. it's huge enough already.